### PR TITLE
Naming and null checking consistency in Web Encoders

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Made the code more consistent in:
1. Argument null checking. Now public APIs check for null and throw ArgumentNullException. Non-public APIs use Debug.Assert.
2. "Count" vs "Length" terminology consistency. Length is used for array-like buffers. Count for everything else.
3. "Char" vs "Character" consistency. I use "character" unless it refers to the particular "Char" type.
4. Some other small renames to make APIs cleaner.

Keep in mind that I mainly did the cleanup in encoders, not in the Unicode APIs.
We have a separate workitems to clean up the Unicode APIs.

### CODE REVIEW

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (Eyas) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs]: Is there a reason why you didn't do a `git mv` here? Would be preferable to preserve history.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/CodePointFilter.cs [hunk 2]
```diff
@@ -12,14 +12,14 @@ namespace System.Text.Encodings.Web
     /// </summary>
     public sealed class CodePointFilter : ICodePointFilter
     {
-        private AllowedCharsBitmap _allowedCharsBitmap;
+        private AllowedCharactersBitmap _allowedCharacterssBitmap;
```
Comment (Eyas) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/CodePointFilter.cs]: nit: typo `...Chracterss...`

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs]: good point. I will fix that. I used VS refactoring to rename the type.

Comment (akoeplinger) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs]: Git tracks this using a "max. x% of file content changed", so there's no difference.

Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs]: hmm, I tried to git mv and rebase. It seemed to all work, but this still shows as deleted and added file. Maybe too much changed?

Comment (akoeplinger) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharactersBitmap.cs]: Yeah, that was what I wanted to say. It's just a visualization issue, it's not recorded in the repo and tools let you pick the specific percentage when showing history.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 3]
```diff
@@ -26,28 +26,33 @@ public static HtmlEncoder Create(params UnicodeRange[] allowedRanges)
...
     public sealed class DefaultHtmlEncoder : HtmlEncoder
     {
-        private AllowedCharsBitmap _allowedCharsBitmap;
+        private AllowedCharactersBitmap _allowedCharacters;
         internal readonly static DefaultHtmlEncoder Singleton = new DefaultHtmlEncoder(new CodePointFilter(UnicodeRanges.BasicLatin));
 
         public DefaultHtmlEncoder(CodePointFilter filter)
         {
-            _allowedCharsBitmap = filter.GetAllowedCharsBitmap();
+            if(filter == null)
```
Comment (Eilon) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: :lipstick: code formatting

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptStringEncoder.cs [hunk 4]
```diff
@@ -26,27 +26,32 @@ public static JavaScriptEncoder Create(params UnicodeRange[] allowedRanges)
...
     {
-        private AllowedCharsBitmap _allowedCharsBitmap;
+        private AllowedCharactersBitmap _allowedCharacters;
 
         internal readonly static DefaultJavaScriptEncoder Singleton = new DefaultJavaScriptEncoder(new CodePointFilter(UnicodeRanges.BasicLatin));
 
         public DefaultJavaScriptEncoder(CodePointFilter filter)
         {
-            _allowedCharsBitmap = filter.GetAllowedCharsBitmap();
+            if(filter == null)
```
Comment (Eilon) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptStringEncoder.cs]: :lipstick:

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoderExtensions.cs [hunk 5]
```diff
@@ -155,13 +163,17 @@ public static void Encode(this TextEncoder encoder, string value, int startIndex
...
-                    EncodeCore(encoder, substring, charCount - firstIndexToEncode, output);
+                    EncodeCore(encoder, substring, characterCount - firstIndexToEncode, output);
                 }
             }
         }
 
-        public static void Encode(this TextEncoder encoder, char[] value, int startIndex, int charCount, TextWriter output)
+        public static void Encode(this TextEncoder encoder, char[] value, int startIndex, int characterCount, TextWriter output)
         {
+            if(encoder == null)
```
Comment (Eilon) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoderExtensions.cs]: :lipstick:

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UrlEncoder.cs [hunk 6]
```diff
@@ -33,25 +33,30 @@ public sealed class DefaultUrlEncoder : UrlEncoder
...
-        public override int MaxOutputCharsPerInputChar
+        public override int MaxOutputCharactersPerInputCharacter
         {
             get { return 9; }
         }
 
         public DefaultUrlEncoder(CodePointFilter filter)
         {
-            _allowedCharsBitmap = filter.GetAllowedCharsBitmap();
+            if(filter == null)
```
Comment (Eilon) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UrlEncoder.cs]: :lipstick:

**Comment by Eilon** [comment]
Just a few <kbd>Ctrl</kbd>+<kbd>K</kbd> <kbd>Ctrl</kbd>+<kbd>D</kbd> issues; after that :shipit:

**Pull request merged**